### DISCUSSION TITLE
How can i have ability to space equally in vertical/horizontal having alignment options

### DISCUSSION THREAD

**Discussion by gsnalinkode** [clientIO/joint#1902] [Q&A]
How can i space equally in vertical/horizontal on after using selection alignment options ?

- Want to implement feature in alignment having option that will space equally in horizontally for selected object 
![image](https://user-images.githubusercontent.com/70625179/203059647-5c8c7bdf-cf83-41c6-b877-685f0b2da13e.png)

- Also want to implement feature in alignment having option that will space equally in vertically for selected object 
![image](https://user-images.githubusercontent.com/70625179/203059951-c988d604-b51f-4b32-bf95-6fb35baa5b6e.png)

**Comment by Tharos** [comment]
Hi @gsnalinkode!

If your question is algorithm related, the task is pretty straightforward. To space horizontally:

1. Calculate the overal width of a place you want to fill (determined by outer elements).
2. Sum up all widths of all elements you want to position.
3. Subtract the sum from the overal width and then divide the difference by `(n - 1)`, where `n` is the count of elements you want to position. The result represents width of a single gap.
4. Use the calculated gap size to position the elements.

To space vertically, the approach is analogous. Be aware that the gap can be negative (occurs when the sum from the step 2 is larger than the overal width), but the algorithm can still handle it.

Was this helpful? 😊
